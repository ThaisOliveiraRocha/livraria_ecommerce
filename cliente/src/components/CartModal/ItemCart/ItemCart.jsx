import React, { useState } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Tr, Td, ImageBook, ImageTrash } from "./ItemCart.styles";
import Text from "../../Text";
import { showModal } from "../../../store/actions";
import { withRouter } from "react-router-dom";

export const ItemCart = ({ item, deleteItem }) => {
  return (
    <>
      <Tr>
        <Td>
          <ImageBook src={item.imagem} />
        </Td>
        <Td>
          <Text>{item.titulo}</Text>
        </Td>
        <Td>
          <Text>{item.qtd}</Text>
        </Td>
        <Td>
          <Text>R${item.preco}</Text>
        </Td>
        <Td>
          <ImageTrash
            className="btnLixeira"
            src={require("../../../assets/images/excluir.png")}
            onClick={() => deleteItem(item.titulo)}
          />
        </Td>
      </Tr>
      
    </>
  );
};

ItemCart.propTypes = {
  // bla: PropTypes.string,
};
const mapStateToProps = state => ({
  // item: state.livraria.carrinho
});

const mapDispatchToProps = dispatch => ({
  deleteItem: titulo => {
    dispatch(showModal(titulo, true));
  }
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(ItemCart));
