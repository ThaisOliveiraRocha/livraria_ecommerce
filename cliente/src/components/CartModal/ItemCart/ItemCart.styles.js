import styled from "styled-components";

export const Tr = styled.tr`
  /* display: flex;
  align-items: center;
  width: 100%; */
`;

export const Td = styled.td`
  text-align: center;
  width: 70px;
  font-weight: normal;
`;

export const ImageBook = styled.img`
  width: 40px;
  height: auto;

  margin: 10px;
`;

export const InfoBook = styled.span`
  font-size: 14px;
  margin-right: 5px;
`;

export const Input = styled.input`
  font-size: 14px;
  width: 50px;
  margin-right: 5px;
`;

export const ImageTrash = styled.img`
  width: 20px;
  height: auto;
  cursor: pointer;

  &:active {
    transform: scale(0.9, 0.9);
  }
`;
