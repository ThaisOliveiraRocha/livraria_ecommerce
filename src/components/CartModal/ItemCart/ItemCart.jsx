import React, { useState } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Tr, Td, ImageBook, ImageTrash } from "./ItemCart.styles";
import Text from "../../../components/Text";
import { showModal } from "../../../store/actions";
import RemoveItemModal from "../../RemoveItemModal";

const ItemCart = ({ item, deleteItem }) => {
  return (
    <>
      <Tr>
        <Td>
          <ImageBook src={require(`../../../assets/images/${item.imagem}`)} />
        </Td>
        <Td>
          <Text>{item.titulo}</Text>
        </Td>
        <Td>
          <Text>{item.qtd}</Text>
        </Td>
        <Td>
          <Text>R${item.preco}</Text>
        </Td>
        <Td>
          <ImageTrash
            src={require("../../../assets/images/excluir.png")}
            onClick={() => deleteItem(item.titulo)}
          />
        </Td>
      </Tr>
      
    </>
  );
};

ItemCart.propTypes = {
  // bla: PropTypes.string,
};
const mapStateToProps = state => ({
  // item: state.livraria.carrinho
});

const mapDispatchToProps = dispatch => ({
  deleteItem: titulo => {
    dispatch(showModal(titulo, true));
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(ItemCart);
