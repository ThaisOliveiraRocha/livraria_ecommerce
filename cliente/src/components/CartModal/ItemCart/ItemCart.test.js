import React from "react";
import { shallow } from "enzyme";
import { ItemCart } from "./ItemCart";

const item = {
  imagem: "book1.png",
  titulo: "É Assim Que Acaba",
  genero: "romance",
  autor: "Colleen Hoover",
  ano: "2018",
  situacao: "novo",
  preco: "40.00",
  qtd: 0
};

describe("<ItemCart />", () => {
  it("should render without crashing", () => {
    shallow(<ItemCart item={item} />);
  });
  it("should render basic infos", () => {
    const wrapper = shallow(<ItemCart item={item} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("clicou no botão e chamou a função", () => {
    const mock = jest.fn();
    const wrapper = shallow(<ItemCart item={item} deleteItem={mock} />);

    wrapper.find(".btnLixeira").simulate("click");

    expect(mock).toHaveBeenCalled();
  });
});
