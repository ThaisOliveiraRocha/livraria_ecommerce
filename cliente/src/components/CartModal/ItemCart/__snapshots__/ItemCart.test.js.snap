// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemCart /> should render basic infos 1`] = `
<Fragment>
  <styled.tr>
    <styled.td>
      <styled.img
        src="book1.png"
      />
    </styled.td>
    <styled.td>
      <Text>
        Ã‰ Assim Que Acaba
      </Text>
    </styled.td>
    <styled.td>
      <Text>
        0
      </Text>
    </styled.td>
    <styled.td>
      <Text>
        R$
        40.00
      </Text>
    </styled.td>
    <styled.td>
      <styled.img
        className="btnLixeira"
        onClick={[Function]}
        src="excluir.png"
      />
    </styled.td>
  </styled.tr>
</Fragment>
`;
